ibm/service/kubernetes/resource_ibm_container_addons.go

```diff
@@ -557,6 +574,20 @@ func resourceIBMContainerAddOnsUpdate(d *schema.ResourceData, meta interface{})
 			newList = new(schema.Set)
 		}
 		os := oldList.(*schema.Set)
+		// manage_all_addons was not previously set to false and the user has added additional or upgraded the existing addons, we set managed_addons to hold the old values
+		if !manageAllAddons && len(managed_addons) == 0 {
```

Comment:
This condition would never hit considering the same condition at line no:536 at the start of this function!!